app/controllers/image_shares_controller.rb

```diff
@@ -0,0 +1,33 @@
+class ImageSharesController < ApplicationController
+  include Images::ControllerHelper
+
+  before_action :find_image
+  before_action :validate_email, only: :create
+
+  def new; end
+
+  def create
+    ImageMailer.share_image_email(@image, share_params[:email], share_params[:message]).deliver_now
+    flash[:success] = 'You have successfully shared an image.'
+    redirect_to images_path
+  rescue Net::ProtocolError
+    flash[:danger] = "We can't send email right now, please try again later"
+    redirect_to new_image_image_share_path(@image)
+  end
+
+  private
+
+  def find_image
+    image_by_id(params[:image_id])
+  end
+
+  def validate_email
+    return if share_params[:email].match?(/\A[^@\s]+@([^@\s]+\.)+[^@\s]+\z/)
+    flash[:danger] = "You didn't give a valid email address"
```

Comment:
It's fine this way.. In APM though, we usually write form object, and do validations in the form. For instance, we would create a ShareImageForm, and then the validation error will be added to the form object, so that when we return a 422 and rerender the form, the error will be displayed inline by simpleform. just FYI